protocol/messenger_handler.go

```diff
@@ -1443,10 +1443,38 @@ func (m *Messenger) HandleCommunityCancelRequestToJoin(state *ReceivedMessageSta
 	}
 
 	requestToJoin, err := m.communitiesManager.HandleCommunityCancelRequestToJoin(signer, cancelRequestToJoinProto)
+
+	if err != nil {
+		return err
+	}
+
+	community, err := m.communitiesManager.GetByID(cancelRequestToJoinProto.CommunityId)
+
 	if err != nil {
 		return err
 	}
 
+	senderID := state.CurrentMessageState.Contact.ID
+	senderPubKey, err := common.HexToPubkey(senderID)
+
+	if err != nil {
+		return err
+	}
+
+	if community.HasMember(senderPubKey) {
+		response, err := m.RemoveUserFromCommunity(
+			community.ID(),
+			senderID,
+		)
+		if err != nil {
+			return err
+		}
+
+		if len(response.Communities()) > 0 {
+			state.Response.AddCommunity(response.Communities()[0])
+		}
+	}
```

Comment:
I believe we should compare the clocks of community requests before removing a member from the community. Let's consider the following:

User actions:
RequestToJoin, clock: 1
CancelRequestToJoin, clock: 2
RequestToJoin, clock: 3

The control node comes back online and receives the requests in the following order:
RequestToJoin, clock: 1
RequestToJoin, clock: 3
CancelRequestToJoin, clock: 2

The user would be removed from the community even when they shouldn't be. To mitigate this, we should handle cancel requests only if they are newer than the latest request to join.

Reply:
Sorry, I don't see how it was resolved?